### ISSUE TITLE
npm console crashes neard on open

### ISSUE THREAD

**Issue by N6REJ** [crazy-max/neard#295]
**When starting the node console it crashes node & neard.** both 0.12.9 & 8.4.1 are effected.

os: Win 10 x64 pro  
neard: 1.2.11
logs: https://drive.google.com/open?id=0B31YVf0qLc5gc2JaZk9mNHN4enc
Demo of issue: https://drive.google.com/file/d/0B31YVf0qLc5gakFQbmJPbFAyeEk/view

**Comment by N6REJ** [comment]
it's a known issue but https://github.com/npm/npm/issues/17839 suggests the latest node.js fixes it.

**Comment by crazy-max** [comment]
Ok thanks i will take a look

**Comment by crazy-max** [comment]
Hi,

No issue for me with Neard 1.2.11 and Node.js 0.12.9 / 8.4.1. Everything works fine.
What is your console version ?

**Comment by N6REJ** [comment]
The same one that came with 1.2.11 .. p2
![image](https://user-images.githubusercontent.com/1850089/28522500-d3ae5e74-703d-11e7-9d7b-fc680bbb0cb1.png)

**Comment by N6REJ** [comment]
just fyi its actually 2 issues max,  1) console crashes neard   2) using npm creates that error shown

**Comment by crazy-max** [comment]
According to your `neard-startup.log`, your system PATH has duplicated elements that can interfers with Neard :

`[2017-07-09 16:18:37] # Neard 1.2.11 # DEBUG: Refresh system PATH: C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\ProgramData\ComposerSetup\bin;%SystemRoot%\system32;%SystemRoot%;%SystemRoot%\System32\Wbem;%SYSTEMROOT%\System32\WindowsPowerShell\v1.0\;%NEARD_BINS%;;C:\Program Files\OpenSSH\bin`

Here is the list of paths in your System PATH :

```
c:\program files\imagemagick-7.0.6-q16
c:\programdata\oracle\java\javapath
c:\windows\system32
c:\windows
c:\windows\system32\wbem
c:\windows\system32\windowspowershell\v1.0\
c:\windows\system32
c:\windows
c:\windows\system32\wbem
c:\windows\system32\windowspowershell\v1.0\
f:\neard\bin\apache\apache2.4.25\bin
f:\neard\bin\php\php7.1.7
f:\neard\bin\php\php7.1.7\pear
f:\neard\bin\php\php7.1.7\imagick
f:\neard\tools\composer\composer1.1.2
f:\neard\tools\drush\drush7.3.0
f:\neard\tools\git\git2.9.0\bin
f:\neard\tools\imagemagick\imagemagick6.9.3-10
f:\neard\tools\phpmetrics\phpmetrics1.0.1
f:\neard\tools\phpunit\phpunit4.8.26
f:\neard\tools\python\python2.7.12.3\bin
f:\neard\tools\ruby\ruby2.0.0.p648\bin
f:\neard\tools\wpcli\wpcli0.20.4
f:\neard\tools\yarn\yarn0.24.6
f:\neard\bin\apache\apache2.4.25\bin
f:\neard\bin\php\php7.1.7
f:\neard\bin\php\php7.1.7\pear
f:\neard\bin\php\php7.1.7\imagick
f:\neard\tools\composer\composer1.1.2
f:\neard\tools\drush\drush7.3.0
f:\neard\tools\git\git2.9.0\bin
f:\neard\tools\imagemagick\imagemagick6.9.3-10
f:\neard\tools\phpmetrics\phpmetrics1.0.1
f:\neard\tools\phpunit\phpunit4.8.26
f:\neard\tools\python\python2.7.12.3\bin
f:\neard\tools\ruby\ruby2.0.0.p648\bin
f:\neard\tools\wpcli\wpcli0.20.4
f:\neard\tools\yarn\yarn0.24.6
%NEARD_BINS%
C:\Program Files\Git\cmd
```

I don't know why you put all these paths (`f:\neard\*`) but it's wrong. I suggest to stop Neard and remove in your System PATH the following elements :

```
c:\program files\imagemagick-7.0.6-q16
f:\neard\bin\apache\apache2.4.25\bin
f:\neard\bin\php\php7.1.7
f:\neard\bin\php\php7.1.7\pear
f:\neard\bin\php\php7.1.7\imagick
f:\neard\tools\composer\composer1.1.2
f:\neard\tools\drush\drush7.3.0
f:\neard\tools\git\git2.9.0\bin
f:\neard\tools\imagemagick\imagemagick6.9.3-10
f:\neard\tools\phpmetrics\phpmetrics1.0.1
f:\neard\tools\phpunit\phpunit4.8.26
f:\neard\tools\python\python2.7.12.3\bin
f:\neard\tools\ruby\ruby2.0.0.p648\bin
f:\neard\tools\wpcli\wpcli0.20.4
f:\neard\tools\yarn\yarn0.24.6
f:\neard\bin\apache\apache2.4.25\bin
f:\neard\bin\php\php7.1.7
f:\neard\bin\php\php7.1.7\pear
f:\neard\bin\php\php7.1.7\imagick
f:\neard\tools\composer\composer1.1.2
f:\neard\tools\drush\drush7.3.0
f:\neard\tools\git\git2.9.0\bin
f:\neard\tools\imagemagick\imagemagick6.9.3-10
f:\neard\tools\phpmetrics\phpmetrics1.0.1
f:\neard\tools\phpunit\phpunit4.8.26
f:\neard\tools\python\python2.7.12.3\bin
f:\neard\tools\ruby\ruby2.0.0.p648\bin
f:\neard\tools\wpcli\wpcli0.20.4
f:\neard\tools\yarn\yarn0.24.6
C:\Program Files\Git\cmd
```

**Comment by N6REJ** [comment]
wow, it was a disaster, I've no clue how that happened... I see other things that I'm not sure if they are correct or not..
![image](https://user-images.githubusercontent.com/1850089/28535975-e050cd96-706b-11e7-8dec-60c5b5eb9b5b.png)
and I'm not sure what to put for %NEARD_BINS%.  If it makes a difference neard is on my USB flash drive.
I have git CLI installed ergo C:\Program Files\Git\cmd

**Comment by crazy-max** [comment]
%NEARD_BINS% is automatically generated when you launch Neard.

**Comment by N6REJ** [comment]
so, remove neard_bins & neard_path?

**Comment by crazy-max** [comment]
You can keep them, there are automatically overwritten when you launch Neard.
Cleaning the PATH solves the issue or not ?

**Comment by N6REJ** [comment]
II removed them then just to keep things clean as I've no idea how they got there...
issue still not resolved, it stayed on the screen longer, and this time I saw white text briefly, then it crashed again...
Checking taskmgr i noticed that even though neard has closed none of the services have.
core/tmp folder is empty.
[logs.zip](https://github.com/crazy-max/neard/files/1174040/logs.zip)

**Comment by N6REJ** [comment]
I thought I had found the cause but shutting all the CC services down didn't solve it either.
![image](https://user-images.githubusercontent.com/1850089/28582762-3507d486-712c-11e7-9d6a-b68aedaa504b.png)

**Comment by crazy-max** [comment]
Hum i think you have node / npm somewhere in your Path.
Can you post your env Path from System and User please ?

**Comment by N6REJ** [comment]
ok, so I tried the latest, first time w/o rebooting it crashed all over the place,  and after CLEAN restart it functioned, and npm ran just fine.  npm install uglify-js ran fine.  EXCEPT.. there was postrgres error and it did not start..
![image](https://user-images.githubusercontent.com/1850089/28602317-f2d4381e-7182-11e7-8adc-4055ea1c45de.png)
so, I shut neard down, and attempted restart, it's currently stalled @ start svn.. The alert doesn't come to the front but when I click on neard in taskbar I get 
![image](https://user-images.githubusercontent.com/1850089/28602353-2303bf46-7183-11e7-97de-334bf19d1d6f.png) clicking on "ok" it continues on and give 8/9 running so I get a orange light.
however, node does run..
![image](https://user-images.githubusercontent.com/1850089/28602386-5f0f69d6-7183-11e7-8462-d0198bbb61a3.png)

after shutting neard down the paths are still present.. 
![image](https://user-images.githubusercontent.com/1850089/28602422-935460f2-7183-11e7-8c58-1c2e9b8f7c45.png)
i wonder if this has anything to do with me being AMD x64 & win 10 pro x64?

**Comment by N6REJ** [comment]
oops, forgot to give you logs..
[tmp.zip](https://github.com/crazy-max/neard/files/1175332/tmp.zip)
[logs.zip](https://github.com/crazy-max/neard/files/1175334/logs.zip)

**Comment by crazy-max** [comment]
@N6REJ You confuse the System path and NEARD_BINS paths.

NEARD_BINS envrionment var is in the System PATH environment var and is expanded via %NEARD_BINS%. So everything in NEARD_BINS will be in System PATH (type `set Path` in your command prompt to check).

NEARD_BINS and NEARD_PATH must be present when stopping Neard because theses vars are used to check if you move your Neard folder. So each time Neard is launched, these vars are checked and if there are not valid, a restart is asked by Neard.

So like i said previously :

**System path** must  be something like this :

```
C:\Windows\system32
C:\Windows
C:\Windows\System32\Wbem
C:\Windows\System32\WindowsPowerShell\v1.0\
%NEARD_BINS%
```

And do not bother with NEARD_BINS or NEARD_PATH.

**Comment by N6REJ** [comment]
max, just fyi, I just tried normal "console" in the main menu and running "pear install PHP_CodeSniffer" caused the crash also.

**Comment by N6REJ** [comment]
idk if this is relevant or not.. 
![image](https://user-images.githubusercontent.com/1850089/28740223-f08734de-73c7-11e7-93ef-026ad5100c8c.png)

**Comment by N6REJ** [comment]
fixed with release [1.2.16](https://github.com/crazy-max/neard/releases/tag/v1.2.16)

**Comment by crazy-max** [comment]
Ok good!
